release-notifier/internal/mail/mail.go

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package mail
+
+import (
+	"bytes"
+	"fmt"
+	"net/smtp"
+	"os"
+	"text/template"
+)
+
+const smtpServer = "smtp.gmail.com"
+const smtpPort = "587"
+const mailTemplate = "templates/mail.html.tmpl"
+const recipentMail = "TRACTUSX_MAILINGLIST"
+const senderMailEnv = "DEVSECOPS_NOTIFICATION_EMAIL"
+const senderPassEnv = "DEVSECOPS_NOTIFICATION_EMAIL_PASSWORD"
+
+func SendPSQLRelNotification(newRelease string, alignedRelease string) {
+	var buff bytes.Buffer
+	mimeHeaders := "MIME-version: 1.0;\nContent-Type: text/html; charset=\"UTF-8\";\n\n"
+	buff.Write([]byte(fmt.Sprintf("Subject: Action Required: PostgreSQL New Release (%s) and Update Process\n%s\n\n", newRelease, mimeHeaders)))
+
+	t, _ := template.ParseFiles(mailTemplate)
+	t.Execute(&buff, struct {
+		NewPSQLRelease string
+		AlignedPSQLRelease string
+	}{
+		NewPSQLRelease: newRelease,
+		AlignedPSQLRelease: alignedRelease,
+	})
+
+	sender := os.Getenv(senderMailEnv)
+	password := os.Getenv(senderPassEnv)
+	recipents := []string{os.Getenv(recipentMail)}
+	sendMail(sender, password, recipents, buff.Bytes())
+}
+
+func sendMail(sender string, password string, recipents []string, body []byte) {
```

Comment:
YAGNI: If we do not send to multiple recipents now, we can keep it as a literal string instead of a slice. This will make the casting of os.Getenv... to slice irrelevant.

Reply:
That's basically to cope with external smtp.SendMail() requiring a slice as parameter. I moved casting to sendMail body.